manim/typing.py

```diff
@@ -1,133 +1,462 @@
+Vector2D: TypeAlias = Point2D
+"""``shape: (2,)``
+A 2D vector: ``[float, float]``.
+Normally a function or method which expects a `Vector2D` as a parameter can handle being
+passed a `Vector3D` instead.
 """
 
-FlatBezierPoints: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (N)` An Array of Bezier Handles but flattened `[float, ...]`."""
+Vector2D_Array: TypeAlias = Point2D_Array
+"""``shape: (M, 2)``
+An array of 2D vectors: ``[[float, float], ...]``.
+Normally a function or method which expects a `Vector2D_Array` as a parameter can handle being
+passed a `Vector3D_Array` instead.
+"""
 
-Point2D_Array: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (N,2)` An Array of Points in 2D Space `[[float, float], ...]`.
+Vector3D: TypeAlias = Point3D
+"""``shape: (3,)``
+A 3D vector: ``[float, float, float]``.
+"""
 
-(Please refer to the documentation of the function you are using for further type Information)
+Vector3D_Array: TypeAlias = Point3D_Array
+"""``shape: (M, 3)``
+An array of 3D vectors: ``[[float, float, float], ...]``.
 """
 
-InternalPoint3D_Array: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (N,3)` An Array of Points in 3D Space `[[float, float, float], ...]`.
-This type alias is mostly made available for internal use and only includes the numpy type.
+VectorND: TypeAlias = Union[npt.NDArray[PointDType], tuple[float, ...]]
+"""``shape (N,)``
+An `N`-D vector: ``[float, ...]``.
 """
 
-Point3D_Array: TypeAlias = Union[
-    InternalPoint3D_Array, Tuple[Tuple[float, float, float], ...]
+VectorND_Array: TypeAlias = tuple[VectorND, ...]
+"""``shape (M, N)``
+An array of `N`-D vectors: ``[[float, ...], ...]``.
+"""
+
+RowVector: TypeAlias = Union[npt.NDArray[PointDType], tuple[tuple[float, ...]]]
+"""``shape: (1, N)``
+A row vector: ``[[float, ...]]``.
+"""
+
+ColVector: TypeAlias = Union[npt.NDArray[PointDType], tuple[tuple[float], ...]]
+"""``shape: (N, 1)``
+A column vector: ``[[float], [float], ...]``.
+"""
+
+
+"""
+[CATEGORY]
+Matrix types
+"""
+
+MatrixMN: TypeAlias = Union[npt.NDArray[PointDType], tuple[tuple[float, ...], ...]]
+"""``shape: (M, N)``
+A matrix: ``[[float, ...], [float, ...], ...]``.
+"""
+
+Zeros: TypeAlias = Union[npt.NDArray[PointDType], tuple[tuple[Literal[0], ...], ...]]
+"""``shape: (M, N)``
+A `MatrixMN` filled with zeros, typically created with ``numpy.zeros((M, N))``."""
+
+
+"""
+[CATEGORY]
+Bézier types
+"""
+
+QuadraticBezierPoints: TypeAlias = Union[
+    npt.NDArray[PointDType], tuple[Point3D, Point3D, Point3D]
+]
+"""``shape: (3, 3)``
+A `Point3D_Array` of 3 control points for a single quadratic Bézier curve: ``[[float, float, float], [float, float, float], [float, float, float]]``.
+"""
+
+QuadraticBezierPoints_Array: TypeAlias = Union[
+    npt.NDArray[PointDType], tuple[QuadraticBezierPoints, ...]
+]
+"""``shape: (N, 3, 3)``
+An array of `N` `QuadraticBezierPoints` objects: ``[[[float, float, float], [float, float, float], [float, float, float]], ...]``.
```

Comment:
I did it because I felt weird to also use the math role with PPC for consistency :sweat_smile: but in the end, I did it with both, typing PPC as \text{PPC}.